OMCompiler/Compiler/NFFrontEnd/NFType.mo

```diff
@@ -1358,8 +1358,9 @@ public
       case ARRAY() then sizeOf(ty.elementType) * Dimension.sizesProduct(ty.dimensions);
       case TUPLE() then List.fold(list(sizeOf(t) for t in ty.types), intAdd, 0);
       case COMPLEX(complexTy = ComplexType.EXTERNAL_OBJECT()) then 1;
-      case COMPLEX()
+      case COMPLEX(complexTy = ComplexType.RECORD())
         then ClassTree.foldComponents(Class.classTree(InstNode.getClass(ty.cls)), fold_comp_size, 0);
+      case COMPLEX() then 1;
```

Comment:
Is this really correct? I feel like this will be wrong for e.g. connectors, but I'm not entirely sure what behaviour we want here.

Reply:
i did not consider frontend stuff to be honest, used it only in the backend and there wont be any connectors right? how would you adapt it for connectors?

Reply:
> i did not consider frontend stuff to be honest, used it only in the backend and there wont be any connectors right?

I hope so 😄